### PR TITLE
Resolves ThrusterVisualizer is Obsolete

### PR DESCRIPTION
**Pull request by TemporalOroboros**
## About the PR

Splits the ThrusterVisualizer into a component/system pair.

### TODO:
- [ ] Testing

**Media**

- [ ] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

### CODE REVIEW

File: Content.Client/Shuttles/ThrusterVisualizerSystem.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+        switch (state)
+        {
+            case true:
+                args.Sprite.LayerSetVisible(ThrusterVisualLayers.ThrustOn, true);
+
+                if (AppearanceSystem.TryGetData(uid, ThrusterVisualState.Thrusting, out bool thrusting, args.Component) && thrusting)
+                {
+                    if (args.Sprite.LayerMapTryGet(ThrusterVisualLayers.Thrusting, out _))
+                    {
```
Comment (ElectroJr) [inline Content.Client/Shuttles/ThrusterVisualizerSystem.cs:23]: use the actual out value

File: Content.Client/Shuttles/ThrusterVisualizerSystem.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+public sealed class ThrusterVisualizerSystem : VisualizerSystem<ThrusterVisualsComponent>
+{
+    /// <summary>
+    /// Updates whether or not the thruster is visibly active/thrusting.
+    /// </summary>
+    protected override void OnAppearanceChange(EntityUid uid, ThrusterVisualsComponent comp, ref AppearanceChangeEvent args)
+    {
+        if (args.Sprite == null)
+            return;
+        if(!AppearanceSystem.TryGetData<bool>(uid, ThrusterVisualState.State, out var state, args.Component))
```
Comment (AJCM-git) [inline Content.Client/Shuttles/ThrusterVisualizerSystem.cs:18]: newline

File: Content.Client/Shuttles/ThrusterVisualsComponent.cs [hunk 3]
```diff
@@ -0,0 +1,8 @@
+namespace Content.Client.Shuttles;
+
+/// <summary>
+/// A component that emits a visible exhaust plume if the entity is an active thruster.
+/// Managed by <see cref="ThrusterVisualizerSystem"/>
+/// </summary>
+[RegisterComponent]
+public sealed class ThrusterVisualsComponent : Component {}
```
Comment (AJCM-git) [inline Content.Client/Shuttles/ThrusterVisualsComponent.cs:8]: access

**Review by AJCM-git — CHANGES_REQUESTED** [review]
this has a bug where when you turn a thruster off the propulsion animation starts

**Comment by DrSmugleaf** [comment]
Reopen when you come back to this.

File: Content.Client/Shuttles/ThrusterVisualizerSystem.cs (hunk 2 above)
Comment (TemporalOroboros) [reply Content.Client/Shuttles/ThrusterVisualizerSystem.cs:18]: Merged into upper if statement.

File: Content.Client/Shuttles/ThrusterVisualsComponent.cs (hunk 3 above)
Comment (TemporalOroboros) [reply Content.Client/Shuttles/ThrusterVisualsComponent.cs:8]: Done

**Comment by TemporalOroboros** [comment]
@DrSmugleaf 
Commits tested and pushed.

**Pull request merged**